### PR TITLE
Add FocusAsync overload with preventScroll parameter #24852

### PR DESCRIPTION
**Pull request by KonstantinAlexeevich**
Hello, this is my first PR. I hope I did everything right.

Summary of the changes (Less than 80 chars)
 - Add FocusAsync overload with preventScroll parameter
 - Add E2E tests with browser scroll position assertaion

#24852
https://github.com/dotnet/aspnetcore/issues/24852

Thanks.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=28686) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=28686) before we can accept your contribution.<br/><br/>:x: KonstantinAlexeevich [sign now](https://cla.dotnetfoundation.org/dotnet/aspnetcore?pullRequest=28686)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/aspnetcore?pullRequest=28686) it.</sub>

File: src/Components/Web/src/ElementReferenceExtensions.cs [hunk 1]
```diff
@@ -17,7 +17,19 @@ public static class ElementReferenceExtensions
         /// </summary>
         /// <param name="elementReference">A reference to the element to focus.</param>
         /// <returns>The <see cref="ValueTask"/> representing the asynchronous focus operation.</returns>
-        public static ValueTask FocusAsync(this ElementReference elementReference)
+        public static ValueTask FocusAsync(this ElementReference elementReference) => elementReference.FocusAsync();
```
Comment (pranavkm) [inline src/Components/Web/src/ElementReferenceExtensions.cs:20]: This likely needs to be: 

```suggestion
        public static ValueTask FocusAsync(this ElementReference elementReference) => elementReference.FocusAsync(preventScroll: false);
```

File: src/Components/Web.JS/src/DomWrapper.ts [hunk 2]
```diff
@@ -4,9 +4,9 @@ export const domFunctions = {
   focus,
 };
 
-function focus(element: HTMLElement): void {
+function focus(element: HTMLElement, options?: FocusOptions): void {
```
Comment (pranavkm) [inline src/Components/Web.JS/src/DomWrapper.ts:7]: Could this be `preventScroll: boolean` for now? It avoids having to JSON serialize a C# object. We could always change this if we start supporting additional features on Focus in the future.

File: src/Components/Web/src/ElementReferenceExtensions.cs [hunk 3]
```diff
@@ -17,7 +17,19 @@ public static class ElementReferenceExtensions
...
+
+        /// <summary>
+        /// Gives focus to an element given its <see cref="ElementReference"/>.
+        /// </summary>
+        /// <param name="elementReference">A reference to the element to focus.</param>
+        /// <param name="preventScroll">
+        ///     A Boolean value indicating whether or not the browser should scroll the document to bring the newly-focused element into view.
```
Comment (pranavkm) [inline src/Components/Web/src/ElementReferenceExtensions.cs:27]:
```suggestion
        ///  A <see cref="boolean" /> value indicating whether or not the browser should scroll the document to bring the newly-focused element into view.
```

File: src/Components/Web/src/ElementReferenceExtensions.cs [hunk 4]
```diff
@@ -17,7 +17,19 @@ public static class ElementReferenceExtensions
...
+        ///     A value of false for preventScroll (the default) means that the browser will scroll the element into view after focusing it.
```
Comment (pranavkm) [inline src/Components/Web/src/ElementReferenceExtensions.cs:28]: Btw, doc comments do not preserve any formatting. If you'd like for this to appear in a separate line, you could consider putting this in a `<para>`.

File: src/Components/test/E2ETest/Tests/ComponentRenderingTest.cs [hunk 5]
```diff
@@ -408,7 +408,12 @@ public void CanUseJsInteropToReferenceElements()
         [Fact]
         public void CanUseFocusExtensionToFocusElement()
         {
+            Browser.Manage().Window.Size = new System.Drawing.Size(100, 200);
```
Comment (pranavkm) [inline src/Components/test/E2ETest/Tests/ComponentRenderingTest.cs:411]: @SteveSandersonMS thoughts on testing this?

File: src/Components/Web/src/ElementReferenceExtensions.cs (hunk 1 above)
Comment (KonstantinAlexeevich) [reply src/Components/Web/src/ElementReferenceExtensions.cs]: yes

Comment (KonstantinAlexeevich) [reply src/Components/Web/src/ElementReferenceExtensions.cs]: Yes. Fix it

File: src/Components/Web.JS/src/DomWrapper.ts (hunk 2 above)
Comment (KonstantinAlexeevich) [reply src/Components/Web.JS/src/DomWrapper.ts]: Yes. Fix it

File: src/Components/Web/src/ElementReferenceExtensions.cs (hunk 3 above)
Comment (KonstantinAlexeevich) [reply src/Components/Web/src/ElementReferenceExtensions.cs]: Yes. Fix it

File: src/Components/Web/src/ElementReferenceExtensions.cs (hunk 4 above)
Comment (KonstantinAlexeevich) [reply src/Components/Web/src/ElementReferenceExtensions.cs:29]: Yes. Fix it

File: src/Components/test/E2ETest/Tests/ComponentRenderingTest.cs (hunk 5 above)
Comment (SteveSandersonMS) [reply src/Components/test/E2ETest/Tests/ComponentRenderingTest.cs]: Assuming this logic works, it looks good to me.

**Review by SteveSandersonMS — APPROVED** [review]
This looks great to me. Thanks, @KonstantinAlexeevich!

**Pull request merged**